staging/src/k8s.io/component-base/logs/api/v1/registry.go

```diff
@@ -0,0 +1,98 @@
+/*
```

Comment:
Diff with removed corresponding file:

```diff
ehashman@red-dot:~/src/k8s$ git diff HEAD^ HEAD -M -- staging/src/k8s.io/component-base/logs/registry/registry.go staging/src/k8s.io/component-base/logs/api/v1/registry.go
diff --git a/staging/src/k8s.io/component-base/logs/api/v1/registry.go b/staging/src/k8s.io/component-base/logs/api/v1/registry.go
index 31a8995c0b5..7871b1389b9 100644
--- a/staging/src/k8s.io/component-base/logs/api/v1/registry.go
+++ b/staging/src/k8s.io/component-base/logs/api/v1/registry.go
@@ -21,6 +21,8 @@ import (
        "sort"
 
        "github.com/go-logr/logr"
+
+       "k8s.io/component-base/featuregate"
 )
 
 var logRegistry = newLogFormatRegistry()
@@ -34,6 +36,10 @@ type logFormatRegistry struct {
 // LogFormatFactory provides support for a certain additional,
 // non-default log format.
 type LogFormatFactory interface {
+       // Feature returns the name of the feature that controls
+       // support for the log format, if there is one.
+       Feature() featuregate.Feature
+
        // Create returns a logger with the requested configuration.
        // Returning a flush function for the logger is optional.
        // If provided, the caller must ensure that it is called
```

Reply:
This enables feature checking for individual logging formats.

Reply:
While rebasing onto https://github.com/kubernetes/kubernetes/pull/108995 I tried something else: I added one commit which moves four files unmodified into `api/v1`, then made the necessary changes. This makes the "[component-base: make LoggingConfiguration a single-version API](https://github.com/kubernetes/kubernetes/pull/105797/commits/0735a3fc49a2f1a6e6dc730ad297c35fde23ff8a#)" commit a bit more readable.

The downside is that bisecting can fail one the one commit that doesn't build.

I don't have a strong preference either way and can also squash before merging.

@ehashman : please let me know what you prefer.

Reply:
I have a strong preference for reviewable commits over bisect purity... :)